sw/device/lib/dif/dif_spi_device.c

```diff
@@ -1542,15 +1542,28 @@ dif_result_t dif_spi_device_tpm_write_data(dif_spi_device_handle_t *spi,
   }
   dif_spi_device_tpm_data_status_t status;
   dif_result_t result = dif_spi_device_tpm_get_data_status(spi, &status);
+  uint8_t offset = length & 0x3;  // lower two bits of length
+  uint32_t rdfifo_wdata;
+
   if (result != kDifOk) {
     return result;
   }
   if (DIF_SPI_DEVICE_TPM_FIFO_DEPTH - status.read_fifo_occupancy < length) {
     return kDifOutOfRange;
   }
-  for (int i = 0; i < length; i++) {
+  for (int i = 0; i < length; i += 4) {
+    if (i + 4 > length) {
+      // Send partial
+      rdfifo_wdata = 0;
+      for (int j = 0; j <= offset; j++) {
+        rdfifo_wdata |= buf[i + j] << (8 * j);
+      }
+    } else {
+      // Type casting to uint32_t then fetch
+      rdfifo_wdata = *((uint32_t *)buf + (i >> 2));
```

Comment:
This is unfortunately not safe--We can't upcast to a wider size and use the pointer directly. Instead, probably need something like...
```suggestion
      // Assemble a 32-bit word.
      rdfifo_wdata = (buf[i+3] << 24) | (buf[i+2] << 16) | (buf[i+1] << 8) | buf[i];
```

This assumes the data is little-endian.

An alternative would be to specify the input array as being a uint32_t* and forcing the caller to handle alignment.

Reply:
I think the latter is better (using uint32_t*). Could you revise the DIF after this is merged?